sqlmesh/core/plan/definition.py

```diff
@@ -426,6 +427,12 @@ def _missing_intervals(self) -> t.Dict[t.Tuple[str, str], Intervals]:
                 if new.is_forward_only:
                     new.dev_intervals = new.intervals.copy()
 
+            deployability_index = (
+                DeployabilityIndex.create({s.snapshot_id: s for s in self.snapshots})
```

Comment:
would probably clean things up if create took in both a collection of snapshots and also the dict